compiler-core/src/type_.rs

```diff
@@ -597,6 +597,14 @@ impl ModuleValueConstructor {
             | ModuleValueConstructor::Constant { documentation, .. } => documentation.as_deref(),
         }
     }
+
+    pub fn name(&self) -> Option<&str> {
```

Comment:
Constants have names too, so I don't think this is a well named function. You could inline the match expression rather than trying to come up with an appropriate generic name for this niche behaviour if that's easier.